### PR TITLE
Create CLI Style Guide

### PR DESCRIPTION
**Pull request by leemunroe**
This style guide is to help align internal and external CLI plugins going forward so the experience is consistent for DC/OS users.

### CODE REVIEW

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# DC/OS CLI Style Guide
+
+## Intro
+The DC/OS CLI is the main tool used by developers and operators to interact with DC/OS cluster and services running on the clusters. The teams developing features and CLI plugins should build the CLI per this style guide so that we offer a consistent and usable experience to users of DC/OS.
+
+## Naming Commands
+* Generally top level commands are single nouns e.g. dcos job, dcos service
+* Top level commands are followed by verbs e.g. dcos job create, dcos cluster rename
+* Command names should always be a single, lowercase word without spaces, hyphens, underscores, or other word delimiters
```
Comment (bamarni) [inline STYLEGUIDE.md]: We currently have `dcos auth list-providers`, in cases like this `dcos auth providers list` would be overkill as this is the only command and it is not meant to grow. Can we say that commands might contains a hyphen delimiter?

File: STYLEGUIDE.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+* `-v` and `-vv` for verbose outputs (for verbose and very verbose)
+* `--lines=N` to restrict amount of results
+
+## Arguments
+
+### Positional Arguments
+
+* A positional argument is a bare value, and its position in a list of arguments identifies it
+
+### Named Arguments
```
Comment (bamarni) [inline STYLEGUIDE.md]: I'm not familiar with this term, do you mean "Options" or "Flags"?

File: STYLEGUIDE.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+* A named argument is a (key, value) pair, where the key identifies the value
+* Flags may or may not accept named arguments
+* Flags should have smart defaults when no named argument is provided
+* Named arguments should be preceded by “=” i.e. not a space
```
Comment (bamarni) [inline STYLEGUIDE.md]: This is very common to accept `=` or a space interchangeably in CLIs, we currently support this as this is what libraries do by default.

File: STYLEGUIDE.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+### ✅ Do
+
+``` 
+$ dcos <command> --<flag>=<argument>
+```
+
+### 🚫 Don't
+
+```
+$ dcos <command> --<flag> <argument>
```
Comment (bamarni) [inline STYLEGUIDE.md]: As commented above, I think both are fine and should be accepted.

File: STYLEGUIDE.md (hunk 2 above)
Comment (leemunroe) [reply STYLEGUIDE.md]: Hmm I pulled this from [here](https://pythonconquerstheuniverse.wordpress.com/2010/07/25/command-line-syntax-some-basic-concepts/) when looking to define the difference between arguments, options, flags. What would you suggest? You have more experience with these terms than me.

File: STYLEGUIDE.md (hunk 3 above)
Comment (leemunroe) [reply STYLEGUIDE.md]: Interesting. In that case should we explicitly state that both should be supported?

File: STYLEGUIDE.md (hunk 1 above)
Comment (bamarni) [reply STYLEGUIDE.md]: sounds good

File: STYLEGUIDE.md (hunk 2 above)
Comment (bamarni) [reply STYLEGUIDE.md]: Arguments / flags is enough for the wording. Then a flag may or may not accept a value.

File: STYLEGUIDE.md (hunk 3 above)
Comment (bamarni) [reply STYLEGUIDE.md]: yes makes sense

**Comment by leemunroe** [comment]
@bamarni Made a few changes to address your concerns.

**Pull request merged**